File: src/main/kotlin/com/example/externalDbOperations/ExternalDbHelper.java
package com.example.externalDbOperations;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public abstract class ExternalDbHelper extends Thread
{
    public static final String IO_EXCEPTION_TAG = "IO_EXCEPTION_TAG";
    public static final String BASE_URL = "http://sobos.ssd-linuxpl.com/";

    public HttpURLConnection setConnection(String address)
    {
        HttpURLConnection conn = null;
        try {
            URL url = new URL(address);
            conn = (HttpURLConnection) url.openConnection();
            conn.setDoOutput(true);
            conn.setDoInput(true);
            conn.setRequestMethod("POST");

        }
        catch (IOException e)
        {
            System.out.println("IOException setConnection: "+e.getMessage());
        }
        return conn;
    }

    public String readResult(HttpURLConnection conn)
    {
        InputStream inputStream;
        StringBuilder readResult = new StringBuilder();
        try {
            inputStream = conn.getInputStream();
            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream,"UTF-8"));
            String line;

            while((line=reader.readLine()) != null)
            {
                readResult.append(line);
            }
            inputStream.close(); reader.close();
            conn.disconnect();
        } catch (IOException e) {
            System.out.println("readResult: "+e.getMessage());
        }

        return readResult.toString();
    }

}

File: src/main/kotlin/com/example/externalDbOperations/ExternalDbHelper.kt
package com.example.externalDbOperations

import com.example.OnResultListener
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking
import tornadofx.launch
import tornadofx.runAsync
import tornadofx.runLater
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL
import kotlin.concurrent.thread

abstract class ExternalDbHelper : Thread() {

    lateinit var onResultListener: OnResultListener

    companion object {
        const val BASE_URL = "http://sobos.ssd-linuxpl.com/"
    }

    fun setConnection(address: String?): HttpURLConnection {
        lateinit var conn: HttpURLConnection
        try {
            val url = URL(address)
            conn = url.openConnection() as HttpURLConnection
            conn.doOutput = true
            conn.doInput = true
            conn.requestMethod = "POST"
        } catch (e: IOException) {
            println("IOException setConnection: " + e.message)
        }
        return conn
    }

    fun readResult(conn: HttpURLConnection): String {
        val inputStream: InputStream
        val readResult = StringBuilder()
        try {
            inputStream = conn.inputStream
            val reader = BufferedReader(InputStreamReader(inputStream, "UTF-8"))
            var line: String?
            while (reader.readLine().also { line = it } != null) {
                readResult.append(line)
            }
            inputStream.close()
            reader.close()
            conn.disconnect()
        } catch (e: IOException) {
            println("readResult: " + e.message)
        }
        return readResult.toString()
    }

    fun setOnResultListener(operation: (String) -> Unit) {
        onResultListener = object : OnResultListener {
            override fun onReceive(result: String) {
                runLater {   operation(result)}
            }
        }
    }
}